Yoshi's Island: Implement New Game (#2141)
ArchipelagoMW/Archipelago: worlds/yoshisisland/docs/setup_en.md

# Yoshi's Island Archipelago Randomizer Setup Guide

## Required Software

> - [Archipelago](https://github.com/ArchipelagoMW/Archipelago/releases). Make sure to check the box for `SNI Client - Yoshi's Island Patch Setup`

Reviewer: ```suggestion
- [Archipelago](https://github.com/ArchipelagoMW/Archipelago/releases).
```
You already know this one, just making it so you can put it in the batch commit.

- Hardware or software capable of loading and playing SNES ROM files
    - An emulator capable of connecting to SNI such as:
        - snes9x-rr from: [snes9x rr](https://github.com/gocha/snes9x-rr/releases),
        - BizHawk from: [TASVideos](https://tasvideos.org/BizHawk)

>         NOTE: RetroArch and FXPakPro are not currently supported.

Reviewer: ```suggestion

        NOTE: RetroArch and FXPakPro are not currently supported.
```
Currently, this line appears on the same line as the Bizhawk link.

- Your legally obtained Yoshi's Island English 1.0 ROM file, probably named `Super Mario World 2 - Yoshi's Island (U).sfc`

## Installation Procedures

### Windows Setup

> 1. During the installation of Archipelago, you will have been asked to install the SNI Client. If you did not do this,
>    or you are on an older version, you may run the installer again to install the SNI Client.
> 2. During setup, you will be asked to locate your base ROM file. This is your Yoshi's Island ROM file.

Reviewer: ```suggestion
1. Download and install Archipelago from the link above, making sure to install the most recent version.
2. During generation/patching, you will be asked to locate your base ROM file. This is your Yoshi's Island ROM file.
```
Took these from the KDL3 docs, SNIClient is always included now and rom won't be asked for until you go to generate or patch.

##### snes9x-rr

1. Load your ROM file if it hasn't already been loaded.
2. Click on the File menu and hover on **Lua Scripting**
3. Click on **New Lua Script Window...**
4. In the new window, click **Browse...**
5. Select the connector lua file included with your client
    - Look in the Archipelago folder for `/SNI/lua/x64` or `/SNI/lua/x86` depending on if the
      emulator is 64-bit or 32-bit.
6. If you see an error while loading the script that states `socket.dll missing` or similar, navigate to the folder of
the lua you are using in your file explorer and copy the `socket.dll` to the base folder of your snes9x install.

##### BizHawk

1. Ensure you have the BSNES core loaded. This is done with the main menubar, under:
    - (≤ 2.8) `Config` 〉 `Cores` 〉 `SNES` 〉 `BSNES`
    - (≥ 2.9) `Config` 〉 `Preferred Cores` 〉 `SNES` 〉 `BSNESv115+`
2. Load your ROM file if it hasn't already been loaded.
   If you changed your core preference after loading the ROM, don't forget to reload it (default hotkey: Ctrl+R).
3. Drag+drop the `Connector.lua` file included with your client onto the main EmuHawk window.

>     - Look in the Archipelago folder for `/SNI/lua/x64` or `/SNI/lua/x86` depending on if the
>       emulator is 64-bit or 32-bit. Please note the most recent versions of BizHawk are 64-bit only.

Reviewer: ```suggestion
```
We probably need to look over a lot of docs for this one tbh. The x64/x86 folders only contain the necessary `socket.dll` now, while Connector.lua got unified.

    - You could instead open the Lua Console manually, click `Script` 〉 `Open Script`, and navigate to `Connector.lua`
      with the file picker.

> 

Reviewer: Feel free to resolve. While you're looking at docs, you might want to check and confirm snes9x-nwa compatibility (see #1778). With 2 of the regular choices being incompatible, might be good to add an alternative.

# Yoshi's Island Archipelago Randomizer Setup Guide

## Required Software

- [Archipelago](https://github.com/ArchipelagoMW/Archipelago/releases). Make sure to check the box for `SNI Client - Yoshi's Island Patch Setup`

- Hardware or software capable of loading and playing SNES ROM files
    - An emulator capable of connecting to SNI such as:
        - snes9x-rr from: [snes9x rr](https://github.com/gocha/snes9x-rr/releases),
        - BizHawk from: [TASVideos](https://tasvideos.org/BizHawk)
        NOTE: RetroArch and FXPakPro are not currently supported.

> - Your legally obtained Yoshi's Island ROM file, probably named `Super Mario World 2 - Yoshi's Island (U).sfc`

Reviewer 2: Which version?
